api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationValueTranslatorImpl.java

```diff
@@ -60,6 +58,15 @@ public Type toFhirResource(@Nonnull Obs obs) {
 			if (obs.getConcept() instanceof ConceptNumeric) {
 				ConceptNumeric cn = (ConceptNumeric) obs.getConcept();
 				result.setUnit(cn.getUnits());
+				
+				// only set the coding system if unit conforms to UCUM standard
+				Coding coding = quantityCodingTranslator.toFhirResource(cn);
+				if (coding.hasSystem() && coding.getSystem().equals(UCUM_SYSTEM_URI)) {
+					String codingSystem = coding.getSystem();
+					result.setCode(coding.getCode());
+					result.setSystem(codingSystem);
+					result.getCodeElement().setSystem(codingSystem);
```

Comment:
not sure if CodeElement.system needs to be set as well but figured i'd be thorough about it